### ISSUE TITLE
[FAQ] How to perform multi value regression?  I have two targets！

### ISSUE THREAD

**Issue by Ever-Bright-V** [basf/mamba-tabular#148]
**Context**
Gives some context if needed (environment, system, hardware).

**Describe the task you are trying to achieve.**
A clear and concise description of what the task is.

**Describe the solution you'd like**

**Comment by Ever-Bright-V** [comment]
import numpy as np
import pandas as pd

# 设置随机种子，以便结果可重复
np.random.seed(42)

# 生成模拟数据
num_samples = 100  # 样本数量

# 生成 X（五列特征）
X = np.random.rand(num_samples, 5)  # 100 x 5 的矩阵

# 生成 y（三列标签），可以根据需要设置不同的分布
y1 = (X[:, 0] + X[:, 1] * 2 + np.random.rand(num_samples) * 0.1) > 1  # 标签1
y2 = (X[:, 2] - X[:, 3] + np.random.rand(num_samples) * 0.1) > 0.5  # 标签2
y3 = (X[:, 4] + np.random.rand(num_samples) * 0.5) > 0.75  # 标签3

# 将数据转为 DataFrame 便于查看
df_X = pd.DataFrame(X, columns=[f'feature_{i+1}' for i in range(5)])
df_y = pd.DataFrame({'label_1': y1.astype(int), 'label_2': y2.astype(int), 'label_3': y3.astype(int)})

# 合并特征和标签
df = pd.concat([df_X, df_y], axis=1)

# 打印前5行数据
# print(df.head())

print("df_y.to_numpy()",df_y.to_numpy(),df_y.to_numpy().shape)

regressor = MambularRegressor()
# regressor=ResNetRegressor()


# Fit the model on training data
regressor.fit(df_X, df_y.to_numpy(), max_epochs=100)

# print(regressor.evaluate(df_X, df_y.to_numpy()))

res=regressor.predict(df_X)
print("res",type(res),res.shape,res)
predict==> return only on Target?

**Comment by amabilee** [comment]
To perform multi-target regression (predicting multiple continuous outcomes), you can use models that support multi-output regression, like `scikit-learn's` `MultiOutputRegressor` and a base regressor like `LinearRegression`.

Alternatively, if you want to perform multi-target regression using only `numpy` and `pandas`, you can manually implement the process without relying on external machine learning libraries.

Which approach are you interested in?

**Comment by Ever-Bright-V** [comment]
Thank you for your patient response. The meaning is that mamba regular does not support multi-target regression.

**Comment by Ever-Bright-V** [comment]
@amabilee

**Comment by Ever-Bright-V** [comment]
Thank you！

**Comment by AnFreTh** [comment]
be aware, however, that this is linear regression. I.e. it wont be as powerful as a ML model. If your targets and predictions are tensors of e.g. torch.Size([N, 3]) the MSE loss will calculate the MSE element wise. I could not tell from you initial message, whether the prediction ultimately only had shape (N,1).
